### PR TITLE
Resolve cross domain communication between the app and the embedded editor

### LINKED ISSUE #3800: Make the embedded NR Editor work in a cross domain-setup
### Epic

#3646

### Description

**As a:** 
user

**I want to:**
Be able to use the embedded NR editor while the editor and app domains differ

The issue is easily reproduced in production where the app domain is .flowfuse.com and instance domains are .flowforge.cloud.

In order to access the embedded editor, use chrome, and from the instance overview page, replace the `/overview` path with `/editor`

### Which customers would this be availble to

Everyone - CE/Starter/Team/Enterprise

### Acceptance Criteria

I am able to access and manage the embedded Node-RED editor using any browser.

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by cstns**
## Description

Investigation of current setup

## Related Issue(s)

closes #3800

## Checklist

 - [ ] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

**Review by knolleary — APPROVED** [review]

**Review by hardillb — APPROVED** [review]

File: frontend/src/pages/instance/components/EditorLink.vue [hunk 1]
```diff
@@ -43,15 +43,11 @@ export default {
...
-        return {
-            canUseImmersiveEditor: false
-        }
-    },
     computed: {
         ...mapState('account', ['team', 'teamMembership']),
         isImmersiveEditor () {
-            return this.canUseImmersiveEditor && SemVer.satisfies(SemVer.coerce(this.instance?.meta?.versions?.launcher), '>=2.3.1')
+            return SemVer.satisfies(SemVer.coerce(this.instance?.meta?.versions?.launcher), '>=2.4.1') &&
+                SemVer.satisfies(SemVer.coerce(this.instance?.meta?.versions?.['node-red']), '>=4.0')
```
Comment (hardillb) [inline frontend/src/pages/instance/components/EditorLink.vue:50]: Only question is will this match the existing beta releases?

Comment (cstns) [reply frontend/src/pages/instance/components/EditorLink.vue:50]: Will we be using beta releases in production? I haven't tested the outcome with a beta release but can do so if needed

Comment (cstns) [reply frontend/src/pages/instance/components/EditorLink.vue:50]: quick test:
```sh
satisfies('3.1.0-beta', '>=2.3.1') true
satisfies('3.1.0-beta', '>=3.2') false
satisfies('3.1.0-beta', '>=4.0') false 
```

Comment (knolleary) [reply frontend/src/pages/instance/components/EditorLink.vue:50]: @cstns it is the 4.0 beta that we care about:
```
satisfies('4.0.0-beta.4', '>=4.0') => false
```

However, as this code uses `coerce` as well, then:

```
satisfies(c.coerce('4.0.0-beta.4'), '>=4.0') => true
```

So I think we're okay.

**Pull request merged**